src/core/public/chrome/ui/header/header_breadcrumbs.tsx

```diff
@@ -51,15 +51,27 @@ export function HeaderBreadcrumbs({ appTitle$, breadcrumbs$, breadcrumbsAppendEx
     ),
   }));
 
-  if (breadcrumbsAppendExtension) {
-    const lastCrumb = crumbs[crumbs.length - 1];
-    lastCrumb.text = (
-      <>
-        {lastCrumb.text}
-        <HeaderExtension extension={breadcrumbsAppendExtension.content} display={'inlineBlock'} />
-      </>
-    );
-  }
+  const Breadcrumbs = (
+    <EuiHeaderBreadcrumbs
+      breadcrumbs={crumbs}
+      max={10}
+      data-test-subj="breadcrumbs"
+      className={classNames({
+        'chrHeaderNavBreadcrumbs--withAppendExtension': !!breadcrumbsAppendExtension,
+      })}
+    />
+  );
+
+  if (!breadcrumbsAppendExtension) return Breadcrumbs;
 
-  return <EuiHeaderBreadcrumbs breadcrumbs={crumbs} max={10} data-test-subj="breadcrumbs" />;
+  return (
+    <EuiFlexGroup responsive={false} gutterSize={'none'} className="eui-textTruncate">
+      <EuiFlexItem grow={false} className="eui-textTruncate">
+        {Breadcrumbs}
+      </EuiFlexItem>
+      <EuiFlexItem grow={true} className="chrHeaderNavBreadcrumbs__appendExtensionContainer">
+        <HeaderExtension extension={breadcrumbsAppendExtension.content} />
```

Comment:
In https://github.com/elastic/kibana/pull/82015#pullrequestreview-520546859 we discussed about a future improvement to allow plugins to mutate the breadcrumb. I think it would be easier to migrate to it later if you were altering the last breadcrumb element instead of creating a flexGroup around it.

Would somethink like

```jsx
lastCrumb.text = (
    <EuiFlexGroup responsive={false} gutterSize={'none'} className="eui-textTruncate">
      <EuiFlexItem grow={false} className="eui-textTruncate">
        {lastCrumb.text}
      </EuiFlexItem>
      <EuiFlexItem grow={true} className="chrHeaderNavBreadcrumbs__appendExtensionContainer">
        <HeaderExtension extension={breadcrumbsAppendExtension.content} />
       </EuiFlexItem>
    </EuiFlexGroup>
);
```

solves the truncation on small width, or are you forced to have the extension outside of the breadcrumb element?

Reply:
That's almost have worked: 

![Screenshot 2020-11-10 at 12 26 53](https://user-images.githubusercontent.com/7784120/98668393-1dd84280-2350-11eb-9627-0874ac6da274.png)

Just no `...` after the title :(

Reply:
`text-overflow` is always a pain to use properly. Anyway, as @kibana-design has been requested for review, maybe they could take a look. Else, keeping it as it is would be fine.

Reply:
The reason the `text-overflow: ellipsis;` isn't being honored here is because the `EuiFlexItem` components have a style of `display: flex;`. Changing the offending `EuiFlexItem` to `display: block;` should remedy it (if this is a path you would prefer to take).

Reply:
@MichaelMarcialis, thanks for the tip! it helped and I made it this way.